rir/src/ir/cleanup.h

```diff
@@ -129,20 +172,25 @@ class BCCleanup : public InstructionDispatcher::Receiver {
         }
     }
 
-    // TODO there is some brokennes when there is dead code
-    // void br_(CodeEditor::Iterator ins) override {
-    //     auto target = code_.target(*ins);
-    //     auto cont = target+1;
-    //     if ((*cont).isReturn()) {
-    //         auto cur = ins.asCursor(code_);
-    //         cur.remove();
-    //         cur << *cont;
-    //     }
-    // }
+    void ret_(CodeEditor::Iterator ins) override { removeDead(ins + 1); }
+
+    void return_(CodeEditor::Iterator ins) override { removeDead(ins + 1); }
 
     void run() {
-        for (auto i = code_.begin(); i != code_.end(); ++i)
-            dispatcher.dispatch(i);
+        for (auto i = code_.begin(); i != code_.end(); ++i) {
+            if (!i.deleted())
```

Comment:
maybe you should use a cursor here?

I seem to remember that our idea was, that the iterator always shows you the original code and the cursor shows you the edited code. therefore if you use a cursor, it should skip the deleted BCs

Reply:
ah, but the api of the dispatcher wants an iterator.

I wonder why there could be deleted instructions here? is it that you run it multiple times without comiting the changes?

Reply:
ah, now I understand, you remove more than one instruction, and then the dispatcher catches up, with instructions you already removed.

ok, that sounds reasonable then. maybe add a comment.